src/classes/HH_OppContactRoles_TEST.cls

```diff
@@ -504,5 +504,143 @@ private class HH_OppContactRoles_TEST {
+        Account acct = UTIL_UnitTestData_TEST.listAccT[0];
+        Contact con = UTIL_UnitTestData_TEST.listConT[0];        
+        con.Deceased__c = true;  
+        update con;
+                
+        //create opportunity
+        Opportunity o = new Opportunity(
+            Name='test oppty',
+            CloseDate=date.newInstance(2001,12,12),
+            StageName=UTIL_UnitTestData_TEST.getClosedWonStage(),
+            Amount=100,
+            AccountId=acct.Id,
+            Primary_Contact__c = con.Id
+        );
+        
+        Test.startTest();
+        insert o;
+        Test.stopTest();
+        //check for contact roles
+        List<OpportunityContactRole> createdRoles = [select Role, ContactId, IsPrimary from OpportunityContactRole where OpportunityId = :o.id order by ContactId];
+        System.assertEquals(1, createdRoles.size(), 'The deceased contact should be added as the primary Contact Role for the Opportunity.');
+        System.assertEquals(contactSettingsForTests.npe01__Opportunity_Contact_Role_Default_role__c, createdRoles[0].Role, 'The Role should match the default contact role.');
+        System.assert(createdRoles[0].IsPrimary, 'The Contact Role should be flagged as primary.');
+    }
+
+    /*******************************************************************************************************
+    * @description Move an Opportunity to a different HH Account and verifies primary OCR remains
+    */ 
+    static testmethod void testOppsAcctChangeHHToHH() {
+        if (strTestOnly != '*' && strTestOnly != 'testOppsAcctChangeHHToHH') return;
+        testOppsAcctChange(CAO_Constants.HH_ACCOUNT_TYPE, CAO_Constants.HH_ACCOUNT_TYPE);
+    }
+
+    /*******************************************************************************************************
+    * @description Move an Opportunity to a different One2One Account and verifies primary OCR remains
+    */ 
+    static testmethod void testOppsAcctChangeOne2OneToOne2One() {
+        if (strTestOnly != '*' && strTestOnly != 'testOppsAcctChangeOne2OneToOne2One') return;
+        testOppsAcctChange(CAO_Constants.ONE_TO_ONE_ORGANIZATION_TYPE, CAO_Constants.ONE_TO_ONE_ORGANIZATION_TYPE);
+    }
+
+    /*******************************************************************************************************
+    * @description MMove an Opportunity from Org to HH Account and verifies primary OCR remains
+    */ 
+    static testmethod void testOppsAcctChangeOrgToHH() {
+        if (strTestOnly != '*' && strTestOnly != 'testOppsAcctChangeOrgToHH') return;
+        testOppsAcctChange(null, CAO_Constants.HH_ACCOUNT_TYPE);
+    }
+
+    /*******************************************************************************************************
+    * @description Move an Opportunity from HH to Org Account and verifies primary OCR remains
+    */ 
+    static testmethod void testOppsAcctChangeHHToOrg() {
```

Comment:
* I suggest to annotate test methods with `@IsTest` for better readability.
* Methods should use the `lowerCamelCase` naming convention. Please rename accordingly.